### DISCUSSION TITLE
jhbuild bootstrap-gtk-osx: no such command (did you mean "jhbuild run bootstrap-gtk-osx"?)

### DISCUSSION THREAD

**Discussion by mkdeak** [jralls/gtk-osx-build#110] [Q&A]
I have been trying to install GTK on my Sonoma 14.5 Apple Silicon machine following the steps from: https://wiki.gnome.org/Projects/GTK/OSX/Building. 

I first ran `./gtk-osx-setup.sh` in my home directory, and then I ran

```
jhbuild bootstrap-gtk-osx
```

to which I got the following response:

```
Loading .env environment variables...
I: Modulesets were edited locally but JHBuild is configured to get them from the network, perhaps you need to add use_local_modulesets = True to your /Users/myuser/.config/jhbuildrc.
jhbuild bootstrap-gtk-osx: no such command (did you mean "jhbuild run bootstrap-gtk-osx"?)
```

In particular it seems that jhbuild doesn't recognize the bootstrap-gtk-osx command. What did I do wrong? Do I need to edit any of `jhbuildrc` files? Earlier the `jhbuildrc` that was generated was corrupt (some error message like "Requested host does not match any Subject Alternative Names (SANs) on TLS certificate" was dumped in there.) I removed the jhbuildrc and things got better but now I'm getting this error message. Should I rename `jhbuildrc-custom` to `jhbuildrc`?

**Comment by jralls** [comment]
No, you need to put back the jhbuildrc that gtk-osx-setup.sh installed. It's what provides `bootstrap-gtk-osx` among a bunch of other things that jhbuild needs on macOS.

**Reply by mkdeak** [reply]
Thanks for the reply. I believe my `jhbuildrc` file was somehow corrupted, it only contained some TLS error message "_Requested host does not match any Subject Alternative Names (SANs) on TLS certificate_". Is there a way to undo the changes `gtk-osx-setup.sh` so that I can try again in a clean attempt?

**Reply by jralls** [reply]
You can `rm -rf ~/.new_local ~/.config/jhbuild*` and try again or you can just download jhbuildrc-gtk-osx from here and move it into ~/.config/jhbuildrc. If you set `$DEVPREFIX` to something other that ~/.new_local adjust that first command accordingly.

**Reply by mkdeak** [reply]
I did what you recommended and deleted those directories and reran the `gtk-osx-setup.sh` script. I just checked the resulting `jhbuildrc` file and this all it contains:

```
Requested host does not match any Subject Alternative Names (SANs) on TLS certificate [38c6a12a870ccb1e3455cf49cb196b99145b1e4b10b4d098d851b9ba65219cf7] in use with this connection.

Visit https://docs.fastly.com/en/guides/common-400-errors#error-421-misdirected-request for more information.
```

Do you have any thoughts on what could have gone wrong?

**Reply by jralls** [reply]
Looks like gitlab might have changed the raw URI. Try this: Find BASEURL at the bottom of gtk-osx-setup.sh and change it from
`BASEURL=https://gitlab.gnome.org/GNOME/gtk-osx/raw/master/`
to
`BASEURL=https://gitlab.gnome.org/GNOME/gtk-osx/-/raw/master/`
then re-run gtk-osx-setup.sh after `rm ~/.config/jhbuildrc*`.
